### PR TITLE
Edit new regional partner attributes

### PR DESCRIPTION
**Pull request by mehalshah**
Eventually I think Regional Partners are important enough to deserve their own dashboard. But for now, simplify the way these attributes work

### CODE REVIEW

File: dashboard/app/controllers/regional_partners_controller.rb [hunk 1]
```diff
@@ -119,4 +145,14 @@ def regional_partner_params
   def restricted_users
     User.select(RESTRICTED_USER_ATTRIBUTES_FOR_VIEW)
   end
+
+  def sanitize_date_hash(date_hash)
```
Comment (clareconstantine) [inline dashboard/app/controllers/regional_partners_controller.rb]: I don't see this being called anywhere, so I'm having a hard time picturing what the `date_hash` will look like  - could you give an example? (Even just as a comment on the PR)

File: dashboard/app/models/concerns/serialized_properties.rb [hunk 2]
```diff
@@ -56,6 +56,7 @@ def permitted_params
     end
 
     def define_methods_for_property(property_name)
+      puts property_name
```
Comment (clareconstantine) [inline dashboard/app/models/concerns/serialized_properties.rb]: rogue `puts` statement

**Review by clareconstantine — APPROVED** [review]

File: dashboard/app/controllers/regional_partners_controller.rb (hunk 1 above)
Comment (mehalshah) [reply dashboard/app/controllers/regional_partners_controller.rb]: It's leftover from the old way of doing minimum dates

File: dashboard/app/controllers/regional_partners_controller.rb [hunk 3]
```diff
@@ -53,7 +53,16 @@ def edit
 
   # PATCH /regional_partners/:id
   def update
-    if @regional_partner.update(regional_partner_params)
+    update_params = regional_partner_params.to_h
+    %w(csd csp).each do |course|
+      %w(teacher facilitator).each do |role|
+        %w(open close).each do |state|
+          key = "apps_#{state}_date_#{course}_#{role}".to_sym
+          update_params[key] == Date.parse(regional_partner_params[key]) if regional_partner_params[key].presence
```
Comment (breville) [inline dashboard/app/controllers/regional_partners_controller.rb:61]: Intended to be `=`?

Comment (sureshc) [reply dashboard/app/controllers/regional_partners_controller.rb:61]: oof.  Looks like it.

**Pull request merged**